cmd/tendermint/commands/util.go

```diff
@@ -0,0 +1,160 @@
+package commands
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+)
+
+// indexTaggedStructFields walks through a struct, looking up
+// those fields tagged by the requested tagName e.g:
+//    Foo int `json:"foo"`
+//    Bar string `mapstructure:"bar"`
+// recursively flattening anonymous struct fields by calling indexTaggedStructFields
+// on them.
+func indexTaggedStructFields(v interface{}, tagName string) map[string]interface{} {
```

Comment:
have you tried to search for such func? maybe it exists in go or viper?

Reply:
Not in Go at least, nor in Viper from what I've checked in their godoc https://godoc.org/github.com/spf13/viper